gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java

```diff
@@ -732,16 +739,82 @@ public Builder loadBalancingStrategy(final LoadBalancingStrategy loadBalancingSt
+            }
+            interceptors.add(index + 1, Pair.of(nameOfInterceptor, interceptor));
+
             return this;
         }
 
+        /**
+         * Adds a {@link RequestInterceptor} before another one that will allow manipulation of the {@code HttpRequest}
+         * prior to its being sent to the server.
+         */
+        public Builder addInterceptorBefore(final String subsequentInterceptorName, final String nameOfInterceptor,
+                                            final RequestInterceptor interceptor) {
+            final int index = getInterceptorIndex(subsequentInterceptorName);
+            if (INTERCEPTOR_NOT_FOUND == index) {
+                throw new IllegalArgumentException(subsequentInterceptorName + " interceptor not found");
+            } else if (getInterceptorIndex(nameOfInterceptor) != INTERCEPTOR_NOT_FOUND) {
+                throw new IllegalArgumentException(nameOfInterceptor + " interceptor already exists");
+            } else if (index == 0) {
+                interceptors.addFirst(Pair.of(nameOfInterceptor, interceptor));
+            } else {
+                interceptors.add(index - 1, Pair.of(nameOfInterceptor, interceptor));
+            }
+
+            return this;
+        }
+
+        /**
+         * Adds a {@link RequestInterceptor} to the end of the list that will allow manipulation of the
+         * {@code HttpRequest} prior to its being sent to the server.
+         */
+        public Builder addInterceptor(final String name, final RequestInterceptor interceptor) {
+            if (getInterceptorIndex(name) != INTERCEPTOR_NOT_FOUND) {
+                throw new IllegalArgumentException(name + " interceptor already exists");
+            }
+            interceptors.add(Pair.of(name, interceptor));
+            return this;
+        }
+
+        /**
+         * Removes a {@link RequestInterceptor} from the list. This can be used to remove the default interceptors that
+         * aren't needed.
+         */
+        public Builder removeInterceptor(final String name) {
+            final int index = getInterceptorIndex(name);
+            if (index == INTERCEPTOR_NOT_FOUND) {
+                throw new IllegalArgumentException(name + " interceptor not found");
+            }
+            interceptors.remove(index);
+            return this;
+        }
+
+        private int getInterceptorIndex(final String name) {
+            for (int i = 0; i < interceptors.size(); i++) {
+                if (interceptors.get(i).getLeft().equals(name)) {
+                    return i;
+                }
+            }
+
+            return INTERCEPTOR_NOT_FOUND;
+        }
+
+        /**
+         * Adds an Auth {@link RequestInterceptor} to the end of list of interceptors.
+         */
         public Builder auth(final Auth auth) {
-            interceptors.add(auth);
+            interceptors.add(Pair.of(auth.getClass().getSimpleName().toLowerCase() + "-auth", auth));
             return this;
```

Comment:
Nit: Utilize `addInterceptor` for the slight benefit of the name collision safety check.
```suggestion
            return addInterceptor(auth.getClass().getSimpleName().toLowerCase() + "-auth", auth);
```